lib/floe/workflow.rb

```diff
@@ -84,5 +103,40 @@ def output
     def end?
       context.ended?
     end
+
+    def current_state
+      @states_by_name[context.state_name]
+    end
+
+    private
+
+    def step_nonblock_submit
+      raise "State is already running" if current_state.started?
+
+      start_time = Time.now.utc
+
+      context.execution["StartTime"] ||= start_time
+      context.state["Input"]         ||= context.execution["Input"].dup
```

Comment:
Can we always set input and State (from start_at) at the same time?
I liked the idea of guid at that same time.

we agree execution and state starting time/entered time need to be carefully thought through.

Reply:
1. I prefer if we set input to execution input at the same place we set the initial state
2. I would prefer if we did not set the first state in the workflow#initialize, but that is all wrapped up in passing context into `step` so we can worry about that later.